lib/modules/manager/pip-compile/readme.md

```diff
@@ -12,20 +12,42 @@ You can "activate" the manager by specifying a `fileMatch` pattern such as:
 ```json
 {
   "pip-compile": {
-    "fileMatch": ["(^|/)requirements\\.in$"]
+    "fileMatch": ["(^|/)requirements\\.txt$"]
   }
 }
 ```
 
-### Assumption of `.in`/`.txt`
+### Assumption of header with a command
 
-If Renovate matches/extracts a file, it assumes that the corresponding output file is found by swapping the `.in` for `.txt`.
-e.g. `requirements.in` => `requirements.txt`
-It will not work if files are in separate directories, including `input/requirements.in` and `output/requirements.txt`.
+If Renovate matches a `pip-compile` output file it will extract original command that was used to create it from header in this file. Because of that `pip-compile` manager poses restrictions on how this file is generated:
 
-If no `.in` suffix is found, then a `.txt` suffix is appended for the output file, e.g. `foo.file` would look for a corresponding `foo.file.txt`.
+- Use default header generation, don't use `--no-header` option.
+- Pass all source files explicitly.
 
-We intend to make the mapping configurable in future iterations.
+In turn `pip-compile` manager will find all source files and parse them as package files. Currently only `*.in` files associated with `pip_requirements` manager are handled.
+
+Example header:
+
+```
+#
+# This file is autogenerated by pip-compile with Python 3.11
+# by the following command:
+#
+#    pip-compile --no-emit-index-url --output-file=requirements.txt requirements.in
+#
+```
+
+### Conflicts with other managers
+
+Because `pip-compile` will update source files with their associated manager you should disable them to avoid running these managers twice.
+
+```json
+{
+  "pip_requirements": {
+    "enabled": false
```

Comment:
What if a repository uses both `requirements.in` for its main dependencies, and `requirements.txt` for say, `docs/requirements.txt`? This needs a better suggestion that allows using both at the same time. Does renovate support filepath exclusion for a manager?

Reply:
And massive thanks for your efforts to improve the `pip-compile` manager 🙌

Reply:
I plan to create support to support relative paths and other cases. Please see #26858, as this is the main PR, and this is a draft concerned with indirect deps.